app/models/article.rb

```diff
@@ -730,16 +729,15 @@ def before_destroy_actions
     Articles::BustMultipleCachesWorker.perform_bulk(job_params)
   end
 
-  # TODO: The param can be removed after with FeatureFlag :consistent_rendering
-  def evaluate_front_matter(hash = front_matter)
+  def evaluate_front_matter(hash)
     self.title = hash["title"] if hash["title"].present?
     set_tag_list(hash["tags"]) if hash["tags"].present?
     self.published = hash["published"] if %w[true false].include?(hash["published"].to_s)
 
     self.published_at = hash["published_at"] if hash["published_at"]
     self.published_at ||= parse_date(hash["date"]) if published
 
-    set_main_image
+    set_main_image(hash)
```

Comment:
Also change to accept an argument so that the code would be more explicit. Also `set_main_image` will receive original frontmatter (not from `ContentRenderer`) this way.